backend/app/notify/email.go

```diff
@@ -0,0 +1,162 @@
+package notify
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	log "github.com/go-pkgz/lgr"
+	"github.com/pkg/errors"
+	"gopkg.in/gomail.v2"
+)
+
+// EmailParams contain settings for email set up
+type EmailParams struct {
+	Server    string
+	Port      int
+	Username  string
+	Password  string
+	KeepAlive time.Duration
+}
+
+// Email implements notify.Destination for email
+type Email struct {
+	server    string
+	port      int
+	username  string
+	password  string
+	keepAlive time.Duration
+	sendChan  chan *gomail.Message
+	errChan   chan error
+}
+
+const emailConnectionKeepAlive = 30 * time.Second
+
+//NewEmail makes email object for notifications and run sending daemon
+func NewEmail(ctx context.Context, params EmailParams) (*Email, error) {
+
+	res := Email{
+		server:    params.Server,
+		port:      params.Port,
+		username:  params.Username,
+		password:  params.Password,
+		keepAlive: params.KeepAlive,
+		sendChan:  make(chan *gomail.Message),
+		errChan:   make(chan error),
+	}
+
+	if res.keepAlive == 0 {
+		res.keepAlive = emailConnectionKeepAlive
+	}
+
+	log.Printf(
+		"[DEBUG] create new email notifier for server %s with user %s, keepalive=%s",
+		res.server, res.username, res.keepAlive)
+
+	// Test connection before starting a daemon.
+	tmpConn, err := gomail.NewDialer(res.server, res.port, res.username, res.password).Dial()
+	if err != nil {
+		return &res, errors.Errorf(
+			"[WARN] error establishing test connecting to '%s':%d with username '%s': %s",
+			res.server, res.port, res.username, err)
+	}
+	if err = tmpConn.Close(); err != nil {
+		return &res, errors.Errorf(
+			"[WARN] error closing test connection to %s:%d: %s",
```

Comment:
errors don't need levels, i.e. [WARN] part is for logging only, not error wrapping/reporting

Reply:
It was a mistake from the time it was a log.Printf, fixed now.